### PR TITLE
Add a failing rustdoc-ui test for public infinite recursive type

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
As suggested in https://github.com/rust-lang/rust/pull/110450#discussion_r1169271643.

r? @jyn514

### CODE REVIEW

File: tests/rustdoc-ui/error-in-impl-trait/infinite-recursive-type-2.rs [hunk 1]
```diff
@@ -0,0 +1,3 @@
+pub enum E { //~ ERROR
+    V(E),
+}
```
Comment (jyn514) [inline tests/rustdoc-ui/error-in-impl-trait/infinite-recursive-type-2.rs:3]: I meant exactly the same as the original test, but with both types public:
```suggestion
pub fn f() -> impl Sized {
    pub enum E {
        V(E),
    }

    unimplemented!()
}
```

**Review by jyn514 — COMMENTED** [review]
thank you :heart: do you mind also moving `tests/rustdoc-ui/infinite-recursive-type-impl-trait.rs` to this `error-in-impl-trait` subdirectory while you're at it?

File: tests/rustdoc-ui/error-in-impl-trait/infinite-recursive-type-2.rs (hunk 1 above)
Comment (GuillaumeGomez) [reply tests/rustdoc-ui/error-in-impl-trait/infinite-recursive-type-2.rs:3]: Oh I see. Updating then.

**Comment by jyn514** [comment]
Thank you! r=me with CI passing

**Comment by GuillaumeGomez** [comment]
Didn't even have time to write a message to say it was updated. You're too fast! :laughing:

**Pull request merged**